src/screens/FilterScreen.js

```diff
@@ -1,34 +1,198 @@
+
+
+const ACCESSIBILITY = [
+    {
+        title: accessibility,
+        content: accessibility_options
+    }
+];
+
+const FACILITIES = [
+    {
+        title: facilities,
+        content: facilities_options
+    }
+];
+
+const PRICE = [
+    {
+        title: price,
+        content: price_options
+    }
+];
 
 class FilterScreen extends Component {
 
+    componentWillMount() {
+        const {displaySites} = this.props;
+
+        this.props.navigation.setParams({siteCount: displaySites.length});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const nextSetOfSites = nextProps.displaySites;
+        const currentSites = this.props.displaySites;
+
+
+        if (nextSetOfSites.length !== currentSites.length) {
+            this.props.navigation.setParams({siteCount: nextSetOfSites.length});
+        }
+    }
+
+    static renderRightNavButton = (navigate, {siteCount}) => {
+        if (Platform.OS === 'ios') {
+            return (
+                <Button
+                    title={_.isUndefined(siteCount) ? '' : `${siteCount} Results`}
+                    onPress={() => navigate(navKeys.SEARCH)}
+                    backgroundColor="rgba(0,0,0,0)"
+                    color="rgba(0,122,255,1)"
+                />
+            );
+        } else if (Platform.OS === 'android') {
+            // android-specific code for navigation here
+        }
+    };
+
     static navigationOptions = (props) => {
-        const {navigation: {navigate}} = props;
+        const {navigation: {navigate, state: {params = {}}}} = props;
 
         return {
-            headerTitle: 'Filter Your Search'
+            headerRight: FilterScreen.renderRightNavButton(navigate, params)
         }
 
     };
 
+    renderHeader = (section, index, isActive) => {
+        const {headerStyle, headerTextStyle} = styles;
 
-    render(){
-        return(
-            <Card>
-                <CardSection>
-                    <Text>
-                        Filter your sites
-                    </Text>
-                </CardSection>
-            </Card>
+        return (
+            <View style={headerStyle}>
+                <Text style={headerTextStyle}>
+                    {section.title}
+
+                </Text>
+                <Icon type='entypo' name={isActive ? 'chevron-down' : 'chevron-up'} size={25} color={campsiteIcon}/>
+            </View>
+        );
+    };
+
+    onClickCheckbox = (key) => {
+        this.props.checkboxWasClicked({filterKey: key})
+    };
+
+    renderCheckedState = (key) => {
+        const {filterCriteriaKeys} = this.props;
+        const flatFilterCriteriaList = _(filterCriteriaKeys)
+            .map(categoryKeys => {
+                return _.map(categoryKeys, innerKey => innerKey)
+            })
+            .flatten()
+            .valueOf();
+
+        return _.includes(flatFilterCriteriaList, key);
+    };
```

Comment:
Above we do some simple logic where we basically take all `filterKey`s, flatten the array to have them all in a single list, and see if this specific `CheckBox`'s key matches any of the `filterKey`s. If it does, then it should show as selected. If it doesn't then it should show as unselected. We use this `renderCheckedState` to set the value of the checkboxes below.